lib/aggregate/values/base.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Aggregate
+  module Values
+    class Base
+      def initialize(value)
+        raise "#{self.class.name} could not handle #{value}" unless self.class.handles?(value)
+
+        @value = value
+      end
+
+      def transpose
+        raise "derived types must implement transpose"
+      end
+
+      def handler_transpose(item)
```

Comment:
This kind of feels like it should be a factory.  Like it is doing the work of a factory without being a full fledged factory?

Thoughts?